### DISCUSSION TITLE
terriaJS in angular

### DISCUSSION THREAD

**Discussion by arinmodi** [TerriaJS/terriajs#7068] [Q&A]
I want to integrate the terriaJS library in my angular typescript project, I have installed the terriaJs using `yarn add terriaJS` command, but when I am importing the module, it says `Cannot find module 'terriajs' or its corresponding type declarations.ts`. is it possible to integrate the terriaJS in angular typescript project ? Angular version - 17. 

Thanks.

**Comment by nf-s** [comment]
Hi @arinmodi 

TerriaJS is tightly coupled with React, so it would be difficult to integrate into an existing Angular project.

I would recommend putting a TerriaMap in an iframe. You can then use `postMessage` to communicate with the TerriaMap

- For example https://docs.terria.io/guide/deploying/controlling-in-an-iframe-or-popup/
